wallet/src/signer/trezor_signer/mod.rs

```diff
@@ -131,6 +137,50 @@ impl TrezorSigner {
         }
     }
 
+    /// Attempts to perform an operation on the Trezor client.
+    ///
+    /// If the operation fails due to an I/O error (which may indicate a lost connection to the device),
+    /// the function will attempt to reconnect to the Trezor device once before returning an error.
+    fn perform_trezor_operation<F, R>(
+        &mut self,
+        operation: F,
+        db_tx: &impl WalletStorageReadLocked,
+        key_chain: &impl AccountKeyChains,
+    ) -> SignerResult<R>
+    where
+        F: Fn(&mut Trezor) -> Result<R, trezor_client::Error>,
+    {
+        let mut client = self.client.lock().expect("poisoned lock");
+        let res = operation(&mut client);
+
+        match res {
+            Ok(res) => Ok(res),
+            // In case of a USB IO error try to reconnect, and try again
+            Err(trezor_client::Error::TransportSendMessage(
+                trezor_client::transport::error::Error::Usb(rusb::Error::Io),
+            )) => {
```

Comment:
IMO instead of calling the operation and then checking for error we should just unconditionally call `Trezor::initialize`, passing to it the previosly obtained session_id (it's inside features). At least this is recommended in [the docs](https://docs.trezor.io/trezor-firmware/common/communication/sessions.html).

Reply:
added an init_device(session_id) call which internally calls initialize before each operation.